js_modules/dagit/packages/core/src/resources/ResourceRoot.tsx

```diff
@@ -0,0 +1,208 @@
+import {gql, useQuery} from '@apollo/client';
+import {
+  Box,
+  Colors,
+  Heading,
+  Page,
+  PageHeader,
+  SplitPanelContainer,
+  Table,
+  Tag,
+  Tooltip,
+} from '@dagster-io/ui';
+import * as React from 'react';
+import {useParams} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorFragment';
+import {useTrackPageView} from '../app/analytics';
+import {useDocumentTitle} from '../hooks/useDocumentTitle';
+import {RepositoryLink} from '../nav/RepositoryLink';
+import {SidebarSection} from '../pipelines/SidebarComponents';
+import {Loading} from '../ui/Loading';
+import {repoAddressToSelector} from '../workspace/repoAddressToSelector';
+import {RepoAddress} from '../workspace/types';
+
+import {ResourceRootQuery} from './types/ResourceRoot.types';
+
+interface Props {
+  repoAddress: RepoAddress;
+}
+
+const remapName = (inName: string): string => {
+  if (inName === 'StringSourceType') {
+    return 'String';
+  } else if (inName === 'IntSourceType') {
+    return 'Int';
+  } else if (inName === 'BoolSourceType') {
+    return 'Bool';
+  }
+  return inName;
+};
+
+export const ResourceRoot: React.FC<Props> = (props) => {
+  useTrackPageView();
+
+  const {repoAddress} = props;
+
+  const {resourceName} = useParams<{resourceName: string}>();
+
+  useDocumentTitle(`Resource: ${resourceName}`);
+
+  const resourceSelector = {
+    ...repoAddressToSelector(repoAddress),
+    resourceName,
+  };
+  const queryResult = useQuery<ResourceRootQuery>(RESOURCE_ROOT_QUERY, {
+    variables: {
+      resourceSelector,
+    },
+    fetchPolicy: 'cache-and-network',
+    partialRefetch: true,
+    notifyOnNetworkStatusChange: true,
+  });
+
+  return (
+    <Page>
+      <PageHeader
+        title={
+          <Box flex={{direction: 'row', alignItems: 'center', gap: 12}}>
+            <Heading>{resourceName}</Heading>
+          </Box>
+        }
+      />
+      <Loading queryResult={queryResult} allowStaleData={true}>
+        {({topLevelResourceOrError}) => {
+          if (topLevelResourceOrError.__typename !== 'TopLevelResource') {
+            return null;
```

Comment:
What should this case look like?

Reply:
Added error message display in the error case